ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/reloperators/HiveUnixTimestampSqlOperator.java

```diff
@@ -29,13 +29,9 @@
  */
 public class HiveUnixTimestampSqlOperator {
   public static final SqlFunction INSTANCE =
-      new SqlFunction("UNIX_TIMESTAMP", SqlKind.OTHER_FUNCTION, ReturnTypes.BIGINT, null,
+      new HiveSqlFunction("UNIX_TIMESTAMP", SqlKind.OTHER_FUNCTION, ReturnTypes.BIGINT, null,
           OperandTypes.or(OperandTypes.NILADIC,
               OperandTypes.or(OperandTypes.STRING, OperandTypes.TIMESTAMP, OperandTypes.DATE),
-              OperandTypes.STRING_STRING), SqlFunctionCategory.NUMERIC) {
-        @Override
-        public boolean isDynamicFunction() {
-          return true;
-        }
```

Comment:
Not sure how much we rely on the notion of dynamic function but this PR changes the current behavior. Before the changes we were returning true and after this change we will return false.

Reply:
Yes, you are right, but this is one of the goal of this patch. Otherwise `UNIX_TIMESTAMP` doesn't evaluated during constant folding by Calcite.

Caclcite 1.25
https://github.com/apache/calcite/blob/68b02dfd4af15bc94a91a0cd2a30655d04439555/core/src/main/java/org/apache/calcite/rel/rules/ReduceExpressionsRule.java#L1094-L1099

From Calcite 1.28 treating dynamic functions as constant or not is configurable
https://issues.apache.org/jira/browse/CALCITE-2736

This is a rule level config though.

Reply:
I guess we can't do anything else at the moment. If necessary we can revisit the choice once we upgrade to 1.28+